src/middleware/issueTable.middleware.js

```diff
@@ -0,0 +1,192 @@
+import performanceDbApi from '../services/performanceDbApi.js'
+import logger from '../utils/logger.js'
+import { pagination } from '../utils/pagination.js'
+import { fetchDatasetInfo, fetchEntityCount, fetchIssueEntitiesCount, fetchIssues, fetchLatestResource, fetchOrgInfo, fetchSpecification, formatErrorSummaryParams, isResourceIdNotInParams, logPageError, pullOutDatasetSpecification, reformatIssuesToBeByEntryNumber, takeResourceIdFromParams, validateQueryParams } from './common.middleware.js'
+import { fetchIf, fetchMany, FetchOptions, parallel, renderTemplate } from './middleware.builders.js'
+import * as v from 'valibot'
+
+const paginationPageLength = 50
+
+export const IssueTableQueryParams = v.object({
+  lpa: v.string(),
+  dataset: v.string(),
+  issue_type: v.string(),
+  issue_field: v.string(),
+  pageNumber: v.optional(v.string()),
+  resourceId: v.optional(v.string())
+})
+
+const validateIssueTableQueryParams = validateQueryParams.bind({
+  schema: IssueTableQueryParams
+})
+
+export const setDefaultQueryParams = (req, res, next) => {
+  if (!req.params.pageNumber) {
+    req.params.pageNumber = 1
+  }
+  next()
+}
+
+const fetchEntitiesWithIssues = fetchMany({
+  query: ({ req, params }) => {
+    const pagination = {
+      limit: paginationPageLength,
+      offset: paginationPageLength * (params.pageNumber - 1)
+    }
+    return performanceDbApi.entitiesAndIssuesQuery({
+      resource: req.resource.resource,
+      issueType: req.params.issue_type,
+      issueField: req.params.issue_field,
+      pagination
+    })
+  },
+  result: 'entitiesWithIssues',
+  dataset: FetchOptions.fromParams
+})
+
+/**
+ * Middleware function to prepare issue table template params
+ *
+ * @param {Request} req - Express request object
+ * @param {Response} res - Express response object
+ * @param {NextFunction} next - Next function in the middleware chain
+ */
+export const prepareIssueTableTemplateParams = (req, res, next) => {
+  const { issue_type: issueType, issue_field: issueField, lpa, dataset: datasetId } = req.params
+  const { entitiesWithIssues, specification, pagination, errorSummary } = req
+
+  const tableParams = {
+    columns: specification.fields.map(field => field.field),
+    fields: specification.fields.map(field => field.field),
+    rows: entitiesWithIssues.map((entity, index) => {
+      const columns = {}
+
+      specification.fields.forEach(fieldObject => {
+        const { field } = fieldObject
+        if (field === 'reference') {
+          const pageNumber = index + 1
+          const entityLink = `/organisations/${lpa}/${datasetId}/${issueType}/${issueField}/entry/${pageNumber}`
+          columns[field] = { html: `<a href="${entityLink}">${entity[field]}</a>` }
+        } else if (entity[field]) {
+          columns[field] = { value: entity[field] }
+        } else {
+          columns[field] = { value: '' }
+        }
+      })
+
+      let issues
```

Comment:
You can set `issues` to `{}` by default here. If `JSON.parse()` throws, `Object.entries(issues)` will throw.